arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/TransactionStack.java

```diff
@@ -62,20 +62,27 @@ public void resetBlock() {
     this.relativeTransactionNumber = 0;
   }
 
-  public void enterTransaction(final WorldView world, final Transaction tx, Block block) {
+  public void enterTransaction(
+      final Hub hub, final WorldView world, final Transaction tx, Block block) {
     this.enter();
 
     final TransactionProcessingMetadata newTx =
         new TransactionProcessingMetadata(
-            world, tx, block, relativeTransactionNumber, currentAbsNumber);
+            hub, world, tx, block, relativeTransactionNumber, currentAbsNumber);
 
-    this.txs.add(newTx);
+    transactions.add(newTx);
   }
 
+  // @françois what replaces this for message calls to smart contracts ?
```

Comment:
I don't understand your question. Why do you want to store the CFI of the smart contract recipient of the tx ? We compute it in the recipient AccountFragment. For the moment, the only place where we read the CFI of a deployment transaction is in `TXN_DATA`, this is strange ...

Reply:
> Why do you want to store the CFI of the smart contract recipient of the tx ?

Because the traces require it. The question is when/where does it get computed.

Reply:
we do it at `traceStartTx` in the RomLex so this is ok, already done. We can delete the comment, as this method just set the CFI of the initCode in the TransactionProcessingMetadata. It is read only by `TXN_DATA` for the moment.